nat-lab/tests/mesh_api.py

```diff
@@ -357,24 +358,41 @@ def generate_peer_config(node: Node, allowed_ips: str) -> str:
                 priv_key = server_config["private_key"]
                 commands = [
                     f"echo {priv_key} > /etc/nordlynx/private.key",
-                    "nlx set nordlynx0 private-key /etc/nordlynx/private.key",
+                    "nlx set nordlynx0 private-key /etc/nordlynx/private.key && nlx set nordlynx0 listen-port {server_config['port']}",
                 ]
 
                 for cmd in commands:
-                    os.system(
-                        f"docker exec -d --privileged {server_config['container']} bash -c '{cmd}'"
+                    full_cmd = f"docker exec --privileged {server_config['container']} bash -c '{cmd}'"
+                    ret = os.system(full_cmd)
+                    print(
+                        datetime.now(),
+                        "Executing",
+                        full_cmd,
+                        "on",
+                        server_config["container"],
+                        "with result",
+                        ret,
```

Comment:
Minor: might it be easy to use DockerProcess for that maybe? It already has this print inside of it.